services/impl/src/main/java/org/exoplatform/automatic/translation/rest/AutomaticTranslationRestService.java

```diff
@@ -118,27 +93,44 @@ public Response setActiveConnector(@ApiParam(value = "Connector name", required
   @Path("/setApiKey")
   @RolesAllowed("administrators")
   @Consumes(MediaType.APPLICATION_JSON)
-  @ApiOperation(
-      value = "Set Automatic translation API key",
-      httpMethod = "PUT",
-      response = Response.class,
-      produces = MediaType.APPLICATION_JSON
-  )
-  @ApiResponses(
-      value = {
-          @ApiResponse(code = 200, message = "Request fulfilled"),
-          @ApiResponse(code = 403, message = "Not authorized to change the connector")
-      }
-  )
+  @ApiOperation(value = "Set Automatic translation API key", httpMethod = "PUT", response = Response.class, produces = MediaType.APPLICATION_JSON)
+  @ApiResponses(value = { @ApiResponse(code = 200, message = "Request fulfilled"),
+      @ApiResponse(code = 403, message = "Not authorized to change the connector") })
   public Response setApiKey(@ApiParam(value = "Connector name", required = true) @QueryParam("connector") String connector,
                             @ApiParam(value = "Api Key", required = true) @QueryParam("apikey") String apikey) {
     connector = (connector == null || connector.equals("null")) ? "" : connector;
-    if (automaticTranslationService.setApiKey(connector,apikey)) {
+    if (automaticTranslationService.setApiKey(connector, apikey)) {
       return Response.ok().build();
     } else {
       return Response.status(403).build();
     }
 
   }
-}
 
+  @POST
+  @Path("/translate")
+  @RolesAllowed("users")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "translate message passed in parameter", httpMethod = "GET", response = Response.class, produces = MediaType.APPLICATION_JSON, notes = "This returns the message transalted in the asked locale, by using the active connector")
+  @ApiResponses(value = { @ApiResponse(code = 200, message = "Request fulfilled") })
+  public Response translate(@ApiParam(value = "message", required = true) @FormParam("message") String message,
+                            @ApiParam(value = "locale", required = true) @FormParam("locale") String localeParam) {
+    JSONObject resultJSON = new JSONObject();
```

Comment:
It's better to not instantiate the object here, since it could be unused (memory usage improvement, when a response is returned without a translation message)

Reply:
done